backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/IssuerManager.java

```diff
@@ -162,6 +154,15 @@ public CredDef createCredDef(@NonNull String schemaId, @NonNull String tag, bool
         return result;
     }
 
+    public void deleteCredDef(@NonNull UUID id) {
+        int recs = credExRepo.countIdByCredDefId(id);
```

Comment:
If we have constraints setup correctly the db can handle this as well and should throw an exception.

Reply:
Yes, there does not appear to be many constraints enabled. and I know we weren't using foreign keys, do we want to start using constraints and  foreign keys? I will put an issue in for us to raise and discuss.